core/eolearn/core/eoexecution.py

```diff
@@ -16,25 +16,34 @@
 """
 import concurrent.futures
 import datetime as dt
+import inspect
 import logging
 import threading
 import warnings
 from dataclasses import dataclass
-from logging import Filter, Handler, Logger
-from typing import Any, Callable, Dict, List, Optional, Sequence, Tuple
+from logging import FileHandler, Filter, Handler, Logger
+from typing import Any, Callable, Dict, List, Optional, Protocol, Sequence, Tuple, Union
 
 import fs
 from fs.base import FS
 
 from .eonode import EONode
 from .eoworkflow import EOWorkflow, WorkflowResults
 from .exceptions import EORuntimeWarning
-from .utils.fs import get_base_filesystem_and_path, get_full_path
+from .utils.fs import get_base_filesystem_and_path, get_full_path, pickle_fs, unpickle_fs
 from .utils.logging import LogFileFilter
 from .utils.parallelize import _decide_processing_type, _ProcessingType, parallelize
 
+
+class _FilesystemHandlerFactoryType(Protocol):
```

Comment:
my brain reads this as **filesystem-handler** factory type
perhaps `_HandlerWithFsFactoryType`? I would even be inclined to drop the `Type` suffix, since I dont see it used that often in python.

Reply:
Ok, renamed to `_HandlerWithFsFactoryType`. I would keep `Type` suffix otherwise it can be confused with normal classes.